turnitintooltwo_user.class.php

```diff
@@ -152,12 +151,18 @@ private function get_pseudo_lastname() {
 
         if ((!isset($userinfo->data) || empty($userinfo->data)) && $config->pseudolastname != 0 && $config->lastnamegen == 1) {
             $uniqueid = strtoupper(strrev(uniqid()));
+
+            $userinfoid = '';
+            if (isset($userinfo->id)) {
+                $userinfoid = $userinfo->id;
+            }
+
             $userinfo = new stdClass();
             $userinfo->userid = $this->id;
             $userinfo->fieldid = $config->pseudolastname;
             $userinfo->data = $uniqueid;
-            if (isset($userinfo->data)) {
-                $userinfo->id = $userinfo->id;
+            if (!empty($userinfoid)) {
+                $userinfo->id = $userinfoid;
```

Comment:
I think the only change required should be changing this:
```
            if (isset($userinfo->data)) {
                $userinfo->id = $userinfo->id;
```

To 
```
            if (isset($userinfo->id)) {
                $userinfo->id = $userinfo->id;
```

As it seems to effectively do the same thing.

Reply:
Thanks for your review @keevan 

Actually, the issue is using same variable for `$userinfo`.
```
if (isset($userinfo->id)) {
                $userinfo->id = $userinfo->id;
```
This doesn't work as the variable `$userinfo` is initialized at line 160.

We need to use this PR way, or change the variable name like other turnitin user class.

https://github.com/turnitin/moodle-plagiarism_turnitin/blob/7c01e79f12f2490a9a18b53f40bba77bfaf02dd8/classes/turnitin_user.class.php#L140

Reply:
Ah re-initialised.  Fun. Okay, this is fine to go in. Though it's still not ideal, it can be cleaned up another time since this is a cherry pick.